hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1109,6 +1136,44 @@ private ClientManager createClientManager(URI fsURI, boolean dtEnabled) throws I
         S3_CLIENT_FACTORY_IMPL, DEFAULT_S3_CLIENT_FACTORY_IMPL,
         S3ClientFactory.class);
 
+    S3ClientFactory clientFactory;
+    S3ClientFactory unecnryptedClientFactory = null;
+    CSEMaterials cseMaterials = null;
+
+    if (isCSEEnabled) {
```

Comment:
this should be pulled out somewhere. goal: PRs should leave the s3aFilesystem class cleaner and clearer than before

Reply:
ack